### PR TITLE
new jrpc package, and tests/rework for rest handler

### LINKED ISSUE #612: add rest api calling package
Need quite a fair bit of boiler plate code, even with generics to send some struct serialized as json and receive a response back over http

This came up during testing the rest api endpoint (#611) so let's make it reusable by anyone and pure/simple library package

### PR DESCRIPTION
**Pull request by ldemailly**
Adding a new `jrpc` json rpc client and server library

Use it in tests and in rest handler

Also changed the returned async objects so they can be used and deserialized

Plus improvements

- [x] change list api
- [x] save file async ID and url
- [x] golang 1.18.5
- [x] don't run the build/tests as root 

fixes #612 

[linter and build changes makes this harder to review than it should, not counting the new features, tricky synchronization etc -- apologies]

### CODE REVIEW

**Comment by ldemailly** [comment]
@wiardvanrij ptal if you have time

File: jrpc/jrpcClient.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		return nil, &FetchError{"deserialization error", code, err, bytes}
+	}
+	if !ok {
+		// can still be "ok" for some callers, they can use the result object as it deserialized as expected.
+		return result, &FetchError{"non ok http result", code, nil, bytes}
+	}
+	return result, nil
+}
+
+// Send fetches the result from url and sending optional payload as a POST, GET if missing.
```
Comment (wuhaoyujerry) [inline jrpc/jrpcClient.go]: nit: should be "sends" not "sending"

Comment (ldemailly) [reply jrpc/jrpcClient.go]: thx, fixed

**Review by wuhaoyujerry — APPROVED** [review]
nice work, thx!

**Comment by ldemailly** [comment]
checking pre3 build and release artifacts once done, and will merge and tag 1.35.0 is all checks out

**Pull request merged**